test/source/tests/settings.ts

```diff
@@ -970,6 +970,55 @@ export const defineSettingsTests = (testVariant: TestVariant, testWithBrowser: T
       await settingsPage.close();
     }));
 
+    ava.default('settings - ensure gracious behavior & ui should remain functional when updating client configuration', testWithBrowser(undefined, async (t, browser) => {
+      const acct = 'test-update@settings.flowcrypt.test';
+      mockBackendData.clientConfigurationByAcctEmail[acct] = keyManagerAutogenRules;
+      const setupPage = await BrowserRecipe.openSettingsLoginApprove(t, browser, acct);
+      await SetupPageRecipe.autoSetupWithEKM(setupPage);
+      const {
+        cryptup_testupdatesettingsflowcrypttest_rules: rules1
+      } = await setupPage.getFromLocalStorage([
+        'cryptup_testupdatesettingsflowcrypttest_rules'
+      ]);
+      const clientConfiguration1 = rules1 as ClientConfiguration;
+      expect(clientConfiguration1.flags).to.eql([
+        'NO_PRV_BACKUP',
+        'ENFORCE_ATTESTER_SUBMIT',
+        'PRV_AUTOIMPORT_OR_AUTOGEN',
+        'PASS_PHRASE_QUIET_AUTOGEN',
+        'DEFAULT_REMEMBER_PASS_PHRASE']);
+      expect(clientConfiguration1.disallow_attester_search_for_domains).to.eql([]);
+      expect(clientConfiguration1.enforce_keygen_algo).to.equal('rsa2048');
+      expect(clientConfiguration1.key_manager_url).to.equal('https://localhost:8001/flowcrypt-email-key-manager');
+      const accessToken = await BrowserRecipe.getGoogleAccessToken(setupPage, acct);
+      await setupPage.close();
+      // Set invalid client configuration and check if it ensures gracious behavior & ui remain functional
+      mockBackendData.clientConfigurationByAcctEmail[acct] = {
+        // flags is required but don't return it (to mock invalid client configuration)
+        key_manager_url: 'https://localhost:8001/flowcrypt-email-key-manager'
+      };
+      const extraAuthHeaders = { Authorization: `Bearer ${accessToken}` };
+      const gmailPage = await browser.newPage(t, TestUrls.mockGmailUrl(), undefined, extraAuthHeaders);
+      await PageRecipe.waitForToastToAppearAndDisappear(gmailPage, 'Failed to update FlowCrypt Client Configuration: Missing client configuration flags.');
+      // Ensure previous client configuration remains same
+      const settingsPage = await browser.newPage(t, TestUrls.extensionSettings(acct));
```

Comment:
I expected to see the same toast here as `accountGetAndUpdateLocalStore` is also called when loading `settings/index.htm` (see `settings/index.ts`) but only saw this 
![image](https://user-images.githubusercontent.com/23335944/205101668-8e5f2309-9a8d-400d-a833-dadda2fd41f3.png)


Also, I see that the client configuration is loaded often when composing a pwd-protected message to get `default_message_expire`. So in case of misconfiguration, the compose module won't get its `{ default_message_expire: 180 }` in case of FES, or a value from FlowCryptAPI even if it is there.
Can you please add a test of composing to a pwd recipient here (when client configuration is corrupt but `default_message_expire` is ok, so we can see what the actual result will be?

Still it's a rather strange excuse to update client configuration -- to get the freshest value of `default_message_expire` -- like this is the most important setting. In case of FES -- FES doesn't even return `default_message_expire` -- so this is simply an excuse. Why not just refresh client configuration on every load of compose frame? Or, on the opposite, never refresh the configuration here. And how to notify the user of corrupt configuration. And how often? These are considerations for discussion with @tomholub

Reply:
I agree. Just updated.
> I expected to see the same toast here as accountGetAndUpdateLocalStore is also called when loading settings/index.htm (see settings/index.ts) but only saw this

Reply:
@tomholub Please confirm this so that I can continue

Reply:
It seems to work ok in our main test cases: `gmail` page and `settings` page.
We can create a separate issue for discussion on updating client configuration in compose boxes and respective tests

Reply:
You mean we can merge this PR now and take care of above in separate issue?
You have other feedbacks in this PR?

Reply:
> You mean we can merge this PR now and take care of above in separate issue?

I think so, yes.